src/components/signup/Signup.jsx

```diff
@@ -0,0 +1,146 @@
+import TwitterIcon from '../../assets/images/twitter-icon.svg';
+import validator from '../../utils/formValidator';
+
+class Signup extends Component {
+  state = {
+    userCredentials: {
+      email: '',
+      password: '',
+      username: '',
+    },
+    validationErrors: '',
+  };
+
+  userInputHandler = event => {
+    const { userCredentials } = this.state;
+    userCredentials[event.target.name] = event.target.value;
+    this.setState({ userCredentials });
+  };
+
+  formValidator = () => {
+    const { userCredentials } = this.state;
+    const validationErrors = validator(userCredentials);
+    this.setState({ validationErrors });
+    if (!validationErrors.length) {
+      this.setState({ validationErrors: '' });
+    }
+  };
+
+  submitForm = event => {
+    event.preventDefault();
+    this.props.signupUser({ ...this.state.userCredentials });
+  };
+
+  render() {
+    const signupButtonValues = {
+      type: 'submit',
+      value: 'GET STARTED',
+      class: 'btn-dark',
+    };
+    const { auth } = this.props;
+    return (
+      <div>
+        <Form loading={auth.isLoading}>
+          <Form.Field>
+            <input
+              className="negative"
+              placeholder="Username"
+              onChange={e => {
+                this.formValidator(), this.userInputHandler(e);
+              }}
+              onBlur={e => {
+                this.formValidator(), this.userInputHandler(e);
+              }}
+              value={this.state.userCredentials.username}
+              name="username"
+              required
+            />
+          </Form.Field>
+
+          <Form.Field>
+            <input
+              placeholder="Email"
+              type="email"
+              onChange={e => {
+                this.formValidator(), this.userInputHandler(e);
+              }}
+              onBlur={e => {
+                this.formValidator(), this.userInputHandler(e);
+              }}
+              value={this.state.userCredentials.email}
+              name="email"
+              required
+            />
+          </Form.Field>
+          <Form.Field>
+            <input
+              placeholder="Password"
+              type="password"
+              onChange={e => {
+                this.formValidator(), this.userInputHandler(e);
+              }}
+              onBlur={e => {
+                this.formValidator(), this.userInputHandler(e);
+              }}
+              value={this.state.userCredentials.password}
+              name="password"
+              required
+            />
+          </Form.Field>
+          {this.state.validationErrors.length > 0 && (
+            <div className="ui negative message">
+              {this.state.validationErrors.map(error => (
+                <p key={error}>{error}</p>
+              ))}
+            </div>
+          )}
+          <Button
+            button={signupButtonValues}
+            signUpClick={e => this.submitForm(e)}
```

Comment:
Events are implicitly passed to an event handler, so we can just pass the event handle directly to the prop. 
```javascript
<Button 
   signUpClick={this.submitForm}
/>